Topics/Tech_Stacks/WebSockets/Web-Sockets.md

```diff
@@ -0,0 +1,169 @@
+# Building Interactive Web Applications with WebSockets
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Understanding WebSockets](#understanding-websockets)
+3. [WebSockets vs HTTP](#websockets-vs-http)
+4. [Advantages and Disadvantages](#advantages-and-disadvantages)
+5. [Setting Up](#setting-up)
+6. [Real-World Use Cases](#real-world-use-cases)
+7. [Conclusion](#conclusion)
+8. [Further Reading](#further-reading)
+
+## Introduction
+In the evolving landscape of web development, the need for real-time, interactive web applications is more prominent than ever. Users expect dynamic experiences with instant updates—be it in chat applications, live sports scores, or stock market feeds. WebSockets represent a significant advancement in web communication, allowing for real-time, full-duplex communication between the client and server. This technology is especially relevant to the CSC301 curriculum as it introduces students to modern web development practices, preparing them for the challenges of developing interactive, user-centric software solutions.
+
+## Understanding WebSockets
+WebSockets are a powerful protocol embedded within the familiar web technology stack, enabling a significant leap in communication capabilities over the Internet. They provide a way to overcome the limitations of the traditional HTTP request-response model, especially when it comes to real-time, low-latency communication.
```

Comment:
Provide a brief explanation of what "full-duplex communication" means for readers who may not be familiar with the term.

Reply:
Yeah makes sense, I've added a simpler explanation for this.